examples/diffusion/textual_inversion_stable_diffusion/detsd/pipeline.py

```diff
@@ -0,0 +1,360 @@
+            is_local_main_process = local_process_index == 0
+
+            device = f"cuda:{process_index}" if distributed is not None else "cuda"
+            pipeline_init_kwargs["device"] = device
+
+            # Instantiate the pipeline and load in any checkpoints by uuid.
+            pipeline = cls(**pipeline_init_kwargs)
+            # Only the local chief worker performs the download.
+            if uuid_list:
+                if is_local_main_process:
+                    paths = pipeline.load_from_uuids(uuid_list)
+                else:
+                    paths = None
+                paths = core_context.distributed.broadcast_local(paths)
+                if not is_local_main_process:
+                    pipeline.load_from_checkpoint_paths(paths)
+
+            if local_checkpoint_paths_list:
+                for path_str in local_checkpoint_paths_list:
+                    path = pathlib.Path(path_str)
+                    pipeline.load_from_checkpoint_paths(
+                        checkpoint_paths=path.parent, learned_embeddings_filename=path.name
+                    )
+
+            # Create the Tensorboard writer.
+            tb_dir = core_context.train.get_tensorboard_path()
+            tb_writer = SummaryWriter(log_dir=tb_dir)
+            # Include relevant __call__ args in the tensorboard tag.
+            important_generation_params = ("prompt", "guidance_scale", "num_inference_steps")
+            tb_tag = "/".join([f"{k}: {call_kwargs[k]}" for k in important_generation_params])
+
+            # Use unique seeds, to avoid repeated images, and add the corresponding generator to the
+            # call_kwargs.
+            seed = main_process_generator_seed + process_index
+            generator = torch.Generator(device=device).manual_seed(seed)
+            call_kwargs["generator"] = generator
+            # Add seed information to the tensorboard tag.
+            tb_tag += f"/seed: {seed}"
+            # Update the call_kwargs with the batch size, if needed.
+            if batch_size > 1:
+                call_kwargs["prompt"] = [call_kwargs["prompt"]] * batch_size
+
+            steps_completed = 0
+            num_generated_imgs = 0
+            img_history = []
+
+            # Restore from a checkpoint, if necessary.
+            if latest_checkpoint is not None:
+                with core_context.checkpoint.restore_path(latest_checkpoint) as path:
+                    with open(path.joinpath("metadata.json"), "r") as f:
+                        metadata_dict = json.load(f)
+                        generator_state_dict = torch.load(
+                            path.joinpath("generator_state_dict.pt"),
+                        )
+                        steps_completed = metadata_dict["steps_completed"]
+                        num_generated_imgs = metadata_dict["num_generated_imgs"]
+                        generator.set_state(generator_state_dict[device])
```

Comment:
I don't think this is true? Or at least, I think the behavior I have coded up is more desirable, or at least equally reasonable.

The two cases discussed in the link aren't about Forking, which doesn't carry over checkpoint info (just tested). The cases are instead:
1. Pause then resume in Web UI.
2. Continue from Web UI.

The advice in the second bullet point in the link is about the Continue behavior and it would give the following behavior (for example):
1. Start a Trial with `max_length = 100`.
2. After completion, hit Continue and set `max_length = 100`.
3. The Trial starts over at `steps_completed = 0` and runs for another 100 steps. 
4. After completion, you have a Trial that lists `max_length = 100` even though it was actually trained for 200 steps.

I have it coded up so that whatever you put in `max_length` after hitting Continue is the actual total `max_length` that gets put in the config. So instead of the above, w/ this code you'd have
1. Start a Trial with `max_length = 100`.
2. After completion, hit Continue and set `max_length = 100`.
3. The Trial has its `max_length` bumped up to 200 and continues until 200 total steps have been completed.

I (obviously) find my implementation more straightforward, but I suppose I should follow the docs. Or, we should change the docs.

Thoughts?

Reply:
Woops sorry, looks like I misremembered re: forking!  In that case, I agree that this seems reasonable and I'm not sure why the alternative interpretation was used in the docs.  The fact that it's open for interpretation is a bit confusing.  Maybe @rb-determined-ai has an opinion here?

Reply:
@coreystatendet is right, but it's not forking he's talking about, it's continue training that has that behavior.

@garrett361 the real concern with continuing batch numbers in a new experiment is that every experiment has a searcher, and searchers operate in terms of absolute batches, but they're all hardcoded to start counting at `0` for a particular experiment.  So everything in the system operates like that right now.

Not saying it's ideal how it is... but that's why the system works that way right now.  If somebody were to use the current code with hyperparameter search and they also manually configured a source_checkpoint_trial_id, the resulting hyperparameter search would be broken.  If they never use hpsearch, it's not really a big deal.

Reply:
Changed the checkpointing code to follow the docs.